File: server/src/main/scala/kpn/api/common/monitor/MonitorRouteChangePage.scala
package kpn.api.common.monitor

import kpn.api.common.Bounds
import kpn.api.common.changes.details.ChangeKey

case class MonitorRouteChangePage(
  key: ChangeKey,
  groupName: String,
  groupDescription: String,
  comment: Option[String],
  wayCount: Long,
  waysAdded: Long,
  waysRemoved: Long,
  waysUpdated: Long,
  osmDistance: Long,
  bounds: Bounds,
  routeSegmentCount: Long,
  routeSegments: Seq[MonitorRouteSegment],
  newNokSegments: Seq[MonitorRouteNokSegment],
  resolvedNokSegments: Seq[MonitorRouteNokSegment],
  reference: MonitorRouteReferenceInfo,
  happy: Boolean,
  investigate: Boolean
)

File: backend/api/src/main/java/kpn/api/common/monitor/MonitorRouteChangePage.java
package kpn.api.common.monitor;

import kpn.api.common.Bounds;
import kpn.api.common.changes.details.ChangeKey;
import kpn.api.common.monitor.MonitorReferenceInfo;
import kpn.api.common.monitor.MonitorRouteDeviation;
import kpn.api.common.monitor.MonitorRouteSegment;

import java.util.Optional;
import com.google.common.collect.ImmutableList;

public record MonitorRouteChangePage(
  ChangeKey key,
  String groupName,
  String groupDescription,
  Optional<String> comment,
  Long wayCount,
  Long waysAdded,
  Long waysRemoved,
  Long waysUpdated,
  Long osmDistance,
  Bounds bounds,
  Long routeSegmentCount,
  ImmutableList<MonitorRouteSegment> routeSegments,
  ImmutableList<MonitorRouteDeviation> newDeviations,
  ImmutableList<MonitorRouteDeviation> resolvedDeviations,
  MonitorReferenceInfo reference,
  Boolean happy,
  Boolean investigate
) {
}
